app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -378,6 +379,8 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
 
             DefaultBrowserNotificationWorker.setDefaultBrowserNotificationIfNeeded(applicationContext)
         }
+
+        components.core.store.dispatch(InitAction)
```

Comment:
While this `InitAction` will work to reconfigure the search engines it will also reset the current state.
Looks like we'll need a new action to only re-initialize the search engines list.

Reply:
A new action `RefreshSearchEnginesAction` was added to the `BrowserAction` to allow for refreshing search engines when app locale is changed. There are more details here: https://github.com/mozilla-mobile/firefox-android/pull/135